api/reviews.go

```diff
@@ -6,6 +6,34 @@ import (
 	"net/http"
 )
 
+func (api *API) getReviews(writer http.ResponseWriter, request *http.Request) {
+	ctx := request.Context()
+	bookID := mux.Vars(request)["id"]
```

Comment:
What happens if `bookID` is empty? If you check this here you can return `BAD REQUEST` status and indicate exactly what the problem is. Also failing fast saves time and effort - why pass this to the `dataStore` if it's invalid - it's unnecessary work.

**Another point to consider:** 
By not checking it you are relying on/assuming that `dataStore` will handle this. In my opinion it isn't good practice to assume functionality of another external component - IMO you should base your uses of an external dependency exclusively on the behaviour defined in it's interface as this is the contract it's giving you -

` "I accept X and I return Y or an error"`

From your point of view thats all you need to know and you should use it with only that knowledge and avoid writing code with knowledge of it's internal workings. Doing so leads to tightly couple code. If the internals of that component changes in the future it could break this code even though they haven't changed the interface - thats would be bad. By coding against the interface definition only you protect yourself from being caught out by that.

These are of course just my opinions but are practices I try to follow. It might seem OTT and strict but I've personally been bitten by this issue and it can be very annoying to fix. It would be interesting to see what others think though.

Reply:
I've now added these changes in a different [PR](https://github.com/cadmiumcat/books-api/pull/32). This PR(#34) is based on the other (#32)

For some reason I thought they made more sense there, but I don't remember what the reason was 😁… Friday brain.

Reply:
I think it makes sense to have done the changes in the other PR if the changes were introduced there and inherited by this one.

Reply:
Thank you both - the change is now in this PR